sam: -r maschine panics from type assertion

$ sam -r host
panic: interface conversion: io.WriteCloser is *exec.closeOnce, not *os.File

goroutine 1 [running]:
main.connectto(0x7ffffbe2c038, 0xf, 0xc00008a170, 0x0, 0x0)
        /home/fs/go/src/github.com/9fans/go/cmd/sam/io.go:232 +0x476
main.startup(0x7ffffbe2c038, 0xf, 0x0, 0xc0000bff40, 0x1, 0x1, 0xc00008a170, 0x0, 0x0)
        /home/fs/go/src/github.com/9fans/go/cmd/sam/io.go:237 +0xbf
main.main()
        /home/fs/go/src/github.com/9fans/go/cmd/sam/sam.go:97 +0x6f1

Somethat similar to the fixed problem in #53.

--- a/cmd/sam/io.go
+++ b/cmd/sam/io.go
@@ -171,8 +171,8 @@ func closeio(p Posn) {
 	}
 }
 
-var remotefd0 = os.Stdin
-var remotefd1 = os.Stdout
+var remotefd0 IOFile = os.Stdin
+var remotefd1 IOFile = os.Stdout
 
 func bootterm(machine string, argv []string) {
 	if machine != "" {
@@ -182,8 +182,6 @@ func bootterm(machine string, argv []string) {
 		if err := cmd.Start(); err != nil {
 			log.Fatal(err)
 		}
-		remotefd0.Close()
-		remotefd1.Close()
 		if err := cmd.Wait(); err != nil {
 			log.Fatalf("samterm: %v", err)
 		}
@@ -228,8 +226,8 @@ func connectto(machine string, files []string) {
 	if err != nil {
 		log.Fatalf("%s: %v", RX, err)
 	}
-	remotefd0 = stdout.(*os.File)
-	remotefd1 = stdin.(*os.File)
+	remotefd0 = stdout.(IOFile)
+	remotefd1 = stdin.(IOFile)
 }
 
 func startup(machine string, Rflag bool, argv []string, files []string) {

